plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -809,14 +809,18 @@ func (checker *PodAffinityChecker) InterPodAffinityMatches(pod *api.Pod, meta in
 
 	// Check if the current node match the inter-pod affinity scheduling constraints.
 	// Hard inter-pod affinity is not symmetric, check only when affinity.PodAffinity exists.
-	if affinity.PodAffinity != nil {
+	if affinity != nil && affinity.PodAffinity != nil {
```

Comment:
That's not the same. We want to call PodAntiAffinity always (potentially with antiAffinity = nil) not only when AntiAffinity != nil